File: frontend/update_article_details.py
import os
import re

# 设置文件路径
template_path = os.path.join(os.getcwd(), 'article-detail.html')

# 读取模板文件内容
with open(template_path, 'r', encoding='utf-8') as f:
    template_content = f.read()

# 提取登录模态框部分（从<!-- 登录模态框 -->到<!-- 用户信息模态框 -->）
login_modal_pattern = r'<!-- 登录模态框 -->.*?<!-- 用户信息模态框 -->'
login_modal_content = re.search(login_modal_pattern, template_content, re.DOTALL).group(0)

# 提取JavaScript函数部分（从// 登录/注册相关函数到// 页面加载完成后执行）
js_functions_pattern = r'// 登录/注册相关函数.*?// 页面加载完成后执行'
js_functions_content = re.search(js_functions_pattern, template_content, re.DOTALL).group(0)

# 获取所有article-detail-*.html文件
article_files = [f for f in os.listdir('.') if re.match(r'article-detail-\d+\.html', f)]

# 遍历所有文件并更新
for file_name in article_files:
    print(f'正在更新 {file_name}...')
    file_path = os.path.join(os.getcwd(), file_name)
    
    # 读取当前文件内容
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # 替换登录/注册模态框部分
    content = re.sub(r'<!-- 登录/注册模态框 -->.*?<!-- 用户信息模态框 -->', login_modal_content, content, flags=re.DOTALL)
    
    # 替换JavaScript函数部分
    content = re.sub(r'// 登录/注册相关函数.*?// 页面加载完成后执行', js_functions_content, content, flags=re.DOTALL)
    
    # 保存修改后的文件
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f'已更新 {file_name}')

print('所有article-detail*.html页面已更新完成！')

File: frontend/update_article_details.js
const fs = require('fs');
const path = require('path');

// 读取article-detail.html的内容作为模板
const templatePath = path.join(__dirname, 'article-detail.html');
const templateContent = fs.readFileSync(templatePath, 'utf8');

// 提取需要替换的部分
const loginModalMatch = templateContent.match(/<!-- 登录模态框 -->[\s\S]*?<!-- 注册模态框 -->[\s\S]*?<!-- 用户信息模态框 -->/);
const loginModalContent = loginModalMatch ? loginModalMatch[0] : '';

const jsFunctionsMatch = templateContent.match(/\/\/ 登录\/注册相关函数[\s\S]*?\/\/ 根据URL参数加载文章内容/);
const jsFunctionsContent = jsFunctionsMatch ? jsFunctionsMatch[0] : '';

// 获取所有article-detail*.html文件
const articleDetailFiles = fs.readdirSync(__dirname)
    .filter(file => file.match(/article-detail-\d+\.html$/))
    .map(file => path.join(__dirname, file));

// 遍历所有文件并更新
articleDetailFiles.forEach(file => {
    console.log(`正在更新 ${file}...`);
    let content = fs.readFileSync(file, 'utf8');
    
    // 替换登录/注册模态框部分
    content = content.replace(/<!-- 登录\/注册模态框 -->[\s\S]*?<!-- 用户信息模态框 -->/, loginModalContent);
    
    // 替换JavaScript函数部分
    content = content.replace(/\/\/ 登录\/注册相关函数[\s\S]*?\/\/ 根据URL参数加载文章内容/, jsFunctionsContent);
    
    // 保存修改后的文件
    fs.writeFileSync(file, content, 'utf8');
    console.log(`已更新 ${file}`);
});

console.log('所有article-detail*.html页面已更新完成！');